lib/matplotlib/dates.py

```diff
@@ -426,16 +427,28 @@ def _findall(self, text, substr):
             i = j + 1
         return sites
 
-    # Dalke: I hope I did this math right.  Every 28 years the
-    # calendar repeats, except through century leap years excepting
-    # the 400 year leap years.  But only if you're using the Gregorian
-    # calendar.
+    def strftime_pre_1900(self, dt, fmt, replace_directives=False):
+        """Call time.strftime for years before 1900 by rolling
+        forward a multiple of 28 years.
 
-    def strftime(self, dt, fmt):
-        fmt = self.illegal_s.sub(r"\1", fmt)
-        fmt = fmt.replace("%s", "s")
-        if dt.year > 1900:
-            return cbook.unicode_safe(dt.strftime(fmt))
+        Dalke: I hope I did this math right.  Every 28 years the
+        calendar repeats, except through century leap years excepting
+        the 400 year leap years.  But only if you're using the Gregorian
+        calendar.
+
+        This implementation uses time's strftime and only handles 4-digit
+        years.  If replace_directive is set to True, as a hack we try and
+        replace %y %x %f with the appropriate values.
+        """
+        if replace_directives:
+          replace_map = \
```

Comment:
We support py2.6, so you would have to number the formatters.

On Sun, May 24, 2015 at 3:22 PM, Eric Firing <notifications@github.com>
wrote:

> In lib/matplotlib/dates.py
> <https://github.com/matplotlib/matplotlib/pull/3242#discussion_r30954958>:
>
> > -    def strftime(self, dt, fmt):
> > -        fmt = self.illegal_s.sub(r"\1", fmt)
> > -        fmt = fmt.replace("%s", "s")
> > -        if dt.year > 1900:
> > -            return cbook.unicode_safe(dt.strftime(fmt))
> > +        Dalke: I hope I did this math right.  Every 28 years the
> > +        calendar repeats, except through century leap years excepting
> > +        the 400 year leap years.  But only if you're using the Gregorian
> > +        calendar.
> > +
> > +        This implementation uses time's strftime and only handles 4-digit
> > +        years.  If replace_directive is set to True, as a hack we try and
> > +        replace %y %x %f with the appropriate values.
> > +        """
> > +        if replace_directives:
> > +          replace_map = \
>
> Also, we try to avoid the trailing backslash continuation. How about using
>
> replace_map = dict(f='{:06d}'.format(dt.microsecond),
>                    x='{:02d}/{:02d}/{:02d}'.format(
>                                dt.month, dt.day, dt.year % 100),
>                    y='{:02d}'.format(dt.year % 100))
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/3242/files#r30954958>.
>